File: Models/Part.cs
using Microsoft.Data.SqlClient;
using Rado.Models;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Rado
{
    public class Part : Item
    {
        public Part() { }
        [Required]
        public long PartId { get; set; }
        public long? CarId { get; set; }
        public int? CategoryId { get; set; }
        public int? SubCategoryId { get; set; }
        [Required] 
        public int DealerSubCategoryId { get; set; }
        [Required]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string DealerSubCategoryName { get; set; }
        public int? LeftRightPosition { get; set; }
        public int? FrontBackPosition { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? PartNumber { get; set; }
        public int? ModelId { get; set; }
        public int? EngineType { get; set; }
        public string? EngineModel { get; set; }
        [Required]
        public int? Year { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public int? PowerkWh { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public int? PowerBHP { get; set; }
        [Required]
        public int ModificationId { get; set; }
        public string? Modification { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public int? Millage { get; set; }
        public int? GearboxType { get; set; }
        public int? Bus { get; set; }
        [Required]
        public decimal Price { get; set; }
    }
}

File: Rado/part365/src/app/model/part/part.ts
import { ApprovedType } from '../enum/approveType.enum'

export class Part {
    partId?                 : number
    carId?                  : number
    bus?                    : number
    modelId?                : number
    modificationId?         : number
    categoryId?             : number
    subCategoryId?          : number
    dealerSubCategoryId?    : number
    dealerSubCategoryName?  : string
    description?            : string
    price?                  : number
    powerkWh?               : number
    powerBHP?               : number
    leftRightPosition?      : number
    frontBackPosition?      : number
    count?                  : number
    partNumber?             : string
    engineType?             : number
    engineModel?            : string
    year?                   : number
    millage?                 : number
    gearboxType?            : number
    regionId?               : number
    isCar?                  : boolean
    modifiedTime?           : number
    mainPicture?            : string
    mainImageId?            : number
    approved?               : ApprovedType
}
